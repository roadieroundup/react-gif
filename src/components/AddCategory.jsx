import { useState } from 'react';
import PropTypes from 'prop-types';


export const AddCategory = ({ onNewCategory }) => {
	const [inputValue, setInputValue] = useState('');

	const onInputChange = ({ target }) => {
		setInputValue(target.value);
	};

	const onSubmit = event => {
		event.preventDefault();

		if (inputValue.trim().length <= 1) return;

		// en este contexto categories es lo que tiene dentro el useState
		// setCategories(categories => [inputValue, ...categories]);
		onNewCategory(inputValue.trim());
		setInputValue('');
	};

	return (
		<form onSubmit={onSubmit} aria-label='form'>
			<input
				type="text"
				placeholder="Search a gif"
				value={inputValue}
				onChange={onInputChange}
			/>
		</form>
	);
};


AddCategory.propTypes = {
	onNewCategory: PropTypes.func.isRequired
}