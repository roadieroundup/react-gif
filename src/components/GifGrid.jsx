// import { useEffect, useState } from 'react';
// import { getGifs } from '../helpers/getGifs';
import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types';


export const GifGrid = ({ category }) => {
	const { gifs, isLoading } = useFetchGifs(category);

	// console.log(isLoading)

	// const [gifs, setGifs] = useState([]);

	// useEffect(() => {
	// 	getGifs(category).then(newGifs => setGifs(newGifs));
	// }, []);

	return (
		<>
			<h3>{category}</h3>

			{isLoading && (<h2>Loading...</h2>)}

			<div className="card-grid">
				{gifs.map(image => (
					<GifItem key={image.id} {...image} />
				))}
			</div>
		</>
	);
};

GifGrid.propTypes = {
	category: PropTypes.string.isRequired
}