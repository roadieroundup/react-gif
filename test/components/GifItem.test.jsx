import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('GifItem tests', () => {
	const title = 'Kokun';
	const url = 'http://whatever.gif/';
	test('should match snapshot', () => {
		const { container } = render(<GifItem title={title} url={url} />);

		expect(container).toMatchSnapshot();
	});

	test('should show the gif with the url and the alt given', () => {
		render(<GifItem title={title} url={url} />);
		//mostrar en un punto en especifico
		// screen.debug();

		// expect(screen.getByRole('img').src).toBe(url);
		// expect(screen.getByRole('img').alt).toBe(title);

		const { src, alt } = screen.getByRole('img');
		expect(src).toBe(url);
		expect(alt).toBe(title);
	});

	test('should show the title inside the component', () => {
		render(<GifItem title={title} url={url} />);

		expect(screen.getByText(title)).toBeTruthy();
	});
});
