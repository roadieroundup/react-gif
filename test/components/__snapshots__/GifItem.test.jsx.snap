// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem tests should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Kokun"
      src="http://whatever.gif/"
    />
    <p>
      Kokun
    </p>
  </div>
</div>
`;
